### DISCUSSION TITLE
exclude variable in Detekt Gradle Plugin?

### DISCUSSION THREAD

**Discussion by alexandre-thauvin** [detekt/detekt#5140] [Q&A]
HI,

As I saw in the `1.17.0` release note, the source paths `test/java/` and `test/kotlin/` are now checked. As I see in the [documentation](https://detekt.dev/docs/gettingstarted/gradle/#options-for-detekt-configuration-closure), it is possible to specify the source paths with the source variable. But is there no way to exclude some source paths (e.g. `test/java` and `test/kotlin`) instead of specifying `main/java` and `main/kotlin` ?

I tried this [following solution](https://detekt.dev/docs/gettingstarted/gradle/#leveraging-gradles-sourcetask---excluding-and-including-source-files) with the `exclude("**/test/**)` but my tests are still being checked.

I use the `1.19.0` version and here is my `detekt` task:
```
        detekt {
            buildUponDefaultConfig = true
            autoCorrect = true

            // Construct a baseline filename per module
            def rootDir = new File(rootProject.rootDir, "config/detekt/baselines")
            def projectPath = project.getPath().replaceAll(':', '/')
            def fullPath = new File(rootDir, projectPath)
            baseline = new File(fullPath, "baseline.xml")
            // The directories where detekt looks for source files.
            // Defaults to `files("src/main/java", "src/test/java", "src/main/kotlin", "src/test/kotlin")`.
            // So we need to specify main in order to avoid analyzing **/test/** source path
            source = files(
                    "src/main/kotlin",
                    "src/main/java"
            )
        }
        tasks.withType(io.gitlab.arturbosch.detekt.Detekt).configureEach {
            exclude("**/test/**") // exclude all tests
        }

        tasks.withType(io.gitlab.arturbosch.detekt.DetektCreateBaselineTask).configureEach {
            exclude("**/test/**") // exclude all tests
        }
```

**Comment by cortinico** [comment]
> but my tests are still being checked.

That's correct. A couple of questions:
1. Are you invoking the `detekt` task or `detektMain` (or other task).
2. Are you sure you're not getting tests inside `androidTest` reported instead?
3. Could you provide a small reproducer?

**Reply by alexandre-thauvin** [reply]
1 - I'm using detekt as detektMain doesn't compile > Could not find method detektMain() for arguments [build_eu04fnlijv12pmbmwtt37evwa$_run_closure2$_closure13$_closure16@641fefc8] on project ':apps' of type org.gradle.api.Project.
(I just changed the name of my detekt task to detektMain)
2 - I confirm that it is test and not androidTest that are reported, actually androidTest are not reported.
3 - I will provide a small reproducer later. 

Thanks for the time and the reply!
